### ISSUE TITLE
DMN elements - Scrollable region must have keyboard access

### ISSUE THREAD

**Issue by vpellegrino** [bpmn-io/dmn-js#821]
### Describe the Bug

The hit policy selector and the DMN table are affected by accessibility issue: **Scrollable region must have keyboard access**.

AXE recommendation:
To solve this problem, you need to fix at least (1) of the following:
* Element should have focusable content
* Element should be focusable

![image](https://github.com/bpmn-io/dmn-js/assets/22591802/d9a103d0-1137-433a-8361-ee4b1b1d009e)

### Steps to Reproduce

1. Open the DMN editor (for example on the Web Modeler)
2. Open the hit policy selector
3. Resize the page so that the internal scrollbar appears on both elements (see screenshot)
4. Execute the page scan, by using the ["axe DevTools" extension](https://www.deque.com/axe/devtools/)

### Expected Behavior

No accessibility issue is reported by the axe scan.

### Environment

Please complete the following information:

 - Browser: Chrome
 - OS: macOs
 - Library version: current one (reproducible on https://demo.bpmn.io/dmn/new)

**Comment by nikku** [comment]
Thanks for opening this issue. I guess we'll find some more issues related to a11y if we use axe extensively.

**Comment by nikku** [comment]
Confirmed this to be an actual issue, cf. recording:

----

From the [spec text](https://www.w3.org/WAI/standards-guidelines/act/rules/0ssw9k/proposed/): It's goal is to ensure that "Scrollable content can be reached with sequential focus navigation". Under ["Assumptions"](https://www.w3.org/WAI/standards-guidelines/act/rules/0ssw9k/proposed/#assumptions) the spec writes:

> This rule assumes that all [scrollable elements](https://www.w3.org/WAI/standards-guidelines/act/rules/0ssw9k/proposed/#scrollable-element) with visible content need to be keyboard accessible. [Scrollable elements](https://www.w3.org/WAI/standards-guidelines/act/rules/0ssw9k/proposed/#scrollable-element) that do not need to be keyboard accessible, perhaps because their content is [purely decorative](https://www.w3.org/TR/WCAG22/#dfn-pure-decoration), the scroll area is whitespace, or because scroll can be controlled in some other keyboard accessible way such as through a button or custom scrollbar, may fail this rule but still satisfy [success criterion 2.1.1 Keyboard](https://www.w3.org/TR/WCAG22/#keyboard).

---

What we violate:

* While we are a partial keyboard trap (including `ESC`) the user cannot tab through the entries to select one.

**Comment by barmac** [comment]
The hit policy widget is a [Select-only Combobox](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/). There is already keyboard access to the options via arrow keys. We may want to annotate the control better though so that axe understands the its purpose.

**Comment by nikku** [comment]
@barmac The point is that nothing can be selected using keyboard right now. This is the bug we need to fix.

**Comment by barmac** [comment]
Not on my machine?

https://github.com/bpmn-io/dmn-js/assets/28307541/0e3459e1-d98d-480f-9312-41e2b3fd5a9b

**Comment by barmac** [comment]
This was tested on https://demo.bpmn.io/dmn

We should NOT scroll to the select options via `TAB`.

**Comment by barmac** [comment]
The only issue I see right now from the keyboard perspective is that `TAB` out does not close the options.

**Comment by nikku** [comment]
My expected behavior:

* :a: I can tab through the options to select, and ESC closes the widget
* :b: I cannot tab through the options, use arrow keys only, but then loosing focus shall close the widget.

**Comment by barmac** [comment]
🅱️ is what I'd expect

**Comment by nikku** [comment]
Verified arrow key (only) navigation on select elements. @barmac What about turning it into a simple select? Why do we make it a custom component in the first place?

**Comment by barmac** [comment]
I am OK with plain HTML experience.

**Comment by nikku** [comment]
I'm de-prioritizing this item and adding to our `a11y` list.

**Comment by barmac** [comment]
Fixed via https://github.com/bpmn-io/dmn-js/pull/843

**Comment by nikku** [comment]
Fixed via https://github.com/bpmn-io/dmn-js/commit/6825860ca0e6cd8ff903323e6e3b8ca8687a8243.
